plugins/inputs/prometheus/prometheus.go

```diff
@@ -437,9 +438,23 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 		return fmt.Errorf("%q returned HTTP status %q", u.URL, resp.Status)
 	}
 
-	body, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return fmt.Errorf("error reading body: %w", err)
+	var body []byte
+	if p.ContentLengthLimit != 0 {
+		lr := io.LimitReader(resp.Body, p.ContentLengthLimit+1)
```

Comment:
It seems that io.LimitedReader does not have an interface to determine whether the reading ended due to EOF or because it reached the specified limit. Therefore, I will perform the reading with a limit of the specified Limit + 1 byte and make a determination based on the length of the result.

Reply:
Can you put a summary of this as a comment in the code? So later viewers will see what the purpose was.

Reply:
You can check `lr.N` which holds the number of remaining bytes before reaching the limit, i.e. if `lr.N == 0` we exhausted the limit.